rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -361,6 +361,22 @@ class Subscription : public SubscriptionBase
     }
   }
 
+  // Take the loaned message
+  /*
+   * \param loaned_message : message that the user wants to take
+   */
+  void
+  take_loaded_message(
+    void * loaned_message,
+    const rclcpp::MessageInfo & message_info) override
+  {
+      rcl_ret_t ret = rcl_take_loaned_message(
```

Comment:
The return value shouldn't be ignored.
I recommend to apply the same logic hat is currently found in the executor

and then have the executor use this function rather than dealing with the `rcl_take_loaned_message` directly

Reply:
That's what I assumed the process should look like so we have a more extendable method for the user, while still following DRY principles.